### ISSUE TITLE
Blocking dependency between devtools and webtools

### ISSUE THREAD

**Issue by Tantriss** [phpMv/ubiquity-devtools#12]
Hi, following the documentation about installing in "console mode" ([doc](https://micro-framework.readthedocs.io/en/latest/quickstart/quickstart.html)) I encounter a problem with the command `Ubiquity info:routes`.

Because I installed Ubiquity without the "-a" parameter, the "webtools" package is not installed anywhere and (I'd really like to know why) the "Route" class needed for `info:routes` is in the webtools package.

I don't see a simple way to fix this as it would require some rewrite of devtools and webtools.
I think the major problem here is that devtools is **global** but Ubiquity (the framework) and webtools are **local**.
We can assume that devtools is installed (cause it should be the only recommended way to create an Ubiquity app).
So any components can freely call devtools classes but devtools should never call **locals** classes cause we can't be sure they'll be installed.

To summarise :
- Ubiquity call devtools => good
- webtools call devtools => good
- devtools call [Ubiquity / webtools] => bad / should check before throwing

I hope you understand what I try to say here. ^^

Tantriss.

**Comment by jcheron** [comment]
Hi @Tantriss ,
Thank you for reporting!

The principle of dependencies is simple, but has not yet been described:
- `Ubiquity` must not be dependent on devtools or webtools (it just needs `ubiquity-dev` in development mode)
- `webtools` can use devtools commands (not the classes), provided they do so in a controlled way (by checking their presence).
-  `devtools` never use webtools (because they are not necessarily present)

The classes common to devtools and webtools (therefore used in development mode) are grouped in [ubiquity-dev](https://github.com/phpMv/ubiquity-dev).

If devtools use a class of webtools (such as the `Route` class), It's a mistake: it means that this class should be in ubiquity-dev.

- PR1 in [ubiquity-dev](https://github.com/phpMv/ubiquity-dev): It is therefore necessary to copy the [Route](https://github.com/phpMv/ubiquity-webtools/blob/master/src/Ubiquity/controllers/admin/popo/Route.php) class into ubiquity-dev (respecting the folder tree structure).
- PR2 in [ubiquity-webtools](https://github.com/phpMv/ubiquity-webtools): Delete the webtools `Route` class.
- Check that everything is working correctly (on the devtools side and on the webtools side, by requesting the display of routes).

Can you do these two PRs?
Thank you!
